src/serac/physics/utilities/functional/tests/functional_comparison_L2.cpp

```diff
@@ -127,6 +134,60 @@ TEST(L2, 3D_linear) { functional_test(*mesh3D, L2<1>{}, L2<1>{}, Dimension<3>{})
 TEST(L2, 3D_quadratic) { functional_test(*mesh3D, L2<2>{}, L2<2>{}, Dimension<3>{}); }
 TEST(L2, 3D_cubic) { functional_test(*mesh3D, L2<3>{}, L2<3>{}, Dimension<3>{}); }
 
+/** CUDA workaround
+Issue with std::variant for InputOptions in mesh_utils.hpp
+
+// this file has a lot of warnings
+serac/src/serac/infrastructure/../../serac/physics/utilities/functional/tensor.hpp(347): warning: calling a __host__
+function("std::tuple< ::serac::tensor<double, (int)3 > ,  ::serac::zero > ::operator =") from a __host__ __device__
+function("serac::operator +< ::serac::dual<    ::std::tuple< ::serac::tensor<double, (int)3 > ,  ::serac::zero > > ,
+double, (int)3 > ") is not allowed
```

Comment:
a more human friendly translation of this warning:

`std::tuple`'s operator= is not marked with `__device__`, so it can't be called from a `__device__` context. This is confusing, because nvcc is supposed to support `std::tuple` since 9.2:

https://docs.nvidia.com/cuda/archive/9.2/cuda-toolkit-release-notes/index.html#cuda-compiler-new-features

<img width="798" alt="Screen Shot 2021-06-07 at 4 03 14 PM" src="https://user-images.githubusercontent.com/61714427/121098181-2ee41780-c7aa-11eb-92c7-d19f3bc78557.png">

My guess is that nvcc only supports the `constexpr` parts of `std::tuple` in device code, and since this copy constructor isn't `constexpr` until C++20, it won't work in device code. That being said, if we want to pursue implementing a small `serac::tuple` object, nothing would prevent us from making those constructors `constexpr` and working on GPUs.

Reply:
Thanks. I'll permalink this within the comments itself.